File: ea-domain/src/main/kotlin/com/nbe3/domain/posts/exception/CommentWriterMismatchException.java
package com.nbe3.domain.posts.exception;

import com.nbe3.common.exception.DomainException;

public class CommentWriterMismatchException extends DomainException {
    public static final DomainException EXCEPTION = new CommentWriterMismatchException();

    private CommentWriterMismatchException() {
        super(CommentErrorCode.COMMENT_WRITER_MISMATCH);
    }
}

File: ea-domain/src/main/java/com/nbe2/domain/posts/exception/CommentWriterMismatchException.kt
package com.nbe2.domain.posts.exception

import com.nbe2.common.exception.DomainException

class CommentWriterMismatchException
private constructor(errorCode: CommentErrorCode) : DomainException(errorCode) {
    companion object {
        val EXCEPTION: DomainException =
                CommentWriterMismatchException(
                        CommentErrorCode.COMMENT_WRITER_MISMATCH
                )
    }
}
// object CommentWriterMismatchException : DomainException(CommentErrorCode.COMMENT_WRITER_MISMATCH)
